anvil/src/eth/backend/mem/storage.rs

```diff
@@ -49,13 +63,31 @@ impl InMemoryBlockStates {
         Self {
             states: Default::default(),
             on_disk_states: Default::default(),
-            limit,
-            min_limit: limit.min(10),
+            in_memory_limit: limit,
+            min_in_memory_limit: limit.min(MIN_HISTORY_LIMIT),
+            max_on_disk_limit: MAX_ON_DISK_HISTORY_LIMIT,
+            oldest_on_disk: Default::default(),
             present: Default::default(),
             disk_cache: Default::default(),
         }
     }
 
+    /// This modifies the `limit` what to keep stored in memory.
+    ///
+    /// This will ensure the new limit is between the configured min limit and the
+    /// `DEFAULT_HISTORY_LIMIT` Lower block times will result in `min_limit`.
+    pub fn update_interval_mine_block_time(&mut self, block_time: Duration) {
+        self.min_in_memory_limit = INTERVAL_MINING_HISTORY_LIMIT;
+        let block_time = block_time.as_secs();
+        let max = DEFAULT_HISTORY_LIMIT as u64;
+        // a low block time would result in a lot of additional state (block)
+        let limit = (max.saturating_mul(block_time) / max) as usize;
+
+        self.in_memory_limit = limit.clamp(self.min_in_memory_limit, DEFAULT_HISTORY_LIMIT);
+
+        self.enforce_limits();
```

Comment:
```suggestion
        let block_time = block_time.as_secs();
        let max = DEFAULT_HISTORY_LIMIT as u64;
        // a low block time would result in a lot of additional state (block)
        let limit = (max.saturating_mul(block_time) / max) as usize;

        self.in_memory_limit = limit.clamp(INTERVAL_MINING_HISTORY_LIMIT, DEFAULT_HISTORY_LIMIT);

        self.enforce_limits();
```

Reply:
what's the difference here? hard to see in 8 loc

Reply:
ah nvm, found it

Reply:
but I disagree with this suggestion, since this ignore the currently configured limit